.github/workflows/run-unit-tests.yml

```diff
@@ -1,42 +0,0 @@
-name: Run Unit Tests with Coverage
-
-on:
-  push:
-    branches:
-      - main
-  pull_request:
-    branches:
-      - main
-
-permissions:
-  contents: read
-  checks: write
-  pull-requests: write
-
-jobs:
-  run-unit-tests:
-    runs-on: ubuntu-latest
-
-    steps:
-      - name: Checkout code
-        uses: actions/checkout@v4
-
-      - name: Set up Python
-        uses: actions/setup-python@v5
-        with:
-          python-version: "3.x"
-
-      - name: Install system dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y libopenblas-dev
-
-      - name: Install Python dependencies
-        run: |
-          python -m pip install --upgrade pip
-          pip install .[dev]
-
-      - name: Run unit tests with coverage
-        run: |
-          coverage run -m pytest
```

Comment:
Maybe the `MemoryError` is due to switching from coverage to pytest --cov?

Reply:
ooh boy those are some long logs!

Reply:
I just noticed that on the disk, the log file takes about 2.2 GB, which is definitely a lot.

Reply:
I reduced the number of sources to 100 and ensured that lephare is way less verbose so that we don't create massive log files.